iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -92,380 +71,355 @@ TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionSe
+    iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
+    publisher.offer();
     this->InterOpWait();
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    auto serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
 
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 }
 
 TEST_F(ServiceDiscovery_test, FindSameServiceMultipleTimesReturnsSingleInstance)
 {
     ::testing::Test::RecordProperty("TEST_ID", "21948bcf-fe7e-44b4-b93b-f46303e3e050");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    this->InterOpWait();
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION("service", "instance", "event");
+    iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    auto serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 
-    serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 }
 
-TEST_F(ServiceDiscovery_test, OfferMultiMethodServiceSingleInstance)
+TEST_F(ServiceDiscovery_test, OfferDifferentServicesWithSameInstance)
 {
     ::testing::Test::RecordProperty("TEST_ID", "25bf794d-450e-47ce-a920-ab2ea479af39");
-    EXPECT_TRUE(sut.offerService({"service1", "instance1", "event1"}));
-    EXPECT_TRUE(sut.offerService({"service2", "instance1", "event1"}));
-    EXPECT_TRUE(sut.offerService({"service3", "instance1", "event1"}));
-    this->InterOpWait();
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    const IdString_t INSTANCE = "instance";
```

Comment:
Could be `constexpr`?

Reply:
I think that does not work with the `cxx::string` due to its constructor declaration.